### PR TITLE
support host network mode

### PR DESCRIPTION
**Pull request by tennix**
This PR enables host network mode. This allows all services accessible on host machine. It's useful when using TiKV without TiDB. Especially can be used for TiSpark #18 
PTAL @onlymellb @zhexuany @c4pt0r @lilin90

### CODE REVIEW

File: compose/templates/docker-compose.yml [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
 {{- define "pd_list" }}
-{{- range until (.Values.pd.size | int) }}{{if .}},{{end}}pd{{.}}:2379{{- end }}
+{{- range until (.Values.pd.size | int) }}{{if .}},{{end}}{{if eq $.Values.networkMode "host"}}127.0.0.1:{{add ($.Values.pd.port | int) .}}{{else}}pd{{.}}:2379{{end}}{{- end }}
 {{- end }}
-{{- $grafanaImage := .Values.grafana.image | default "grafana/grafana:4.6.3" }}
-{{- $grafanaPort := .Values.grafana.port | default "3000" }}
-{{- $prometheusImage := .Values.prometheus.image | default "prom/prometheus:v2.0.0" }}
-{{- $prometheusPort := .Values.prometheus.port | default "9090" }}
-{{- $pushgatewayImage := .Values.pushgateway.image | default "prom/pushgateway:v0.3.1" }}
-{{- $dashboardInstallerImage := .Values.dashboardInstaller.image | default "pingcap/tidb-dashboard-installer:v1.0.0" }}
```
Comment (onlymellb) [inline compose/templates/docker-compose.yml]: Why remove these default values?

File: compose/templates/docker-compose.yml [hunk 2]
```diff
@@ -1,17 +1,14 @@
 {{- $pdSize := .Values.pd.size | int }}
 {{- $tikvSize := .Values.tikv.size | int }}
+{{- $pdPort := .Values.pd.port | int }}
+{{- $pdPeerPort := add $pdPort 10000 }}
+{{- $tikvPort := .Values.tikv.port | int }}
 {{- define "initial_cluster" }}
-{{- range until (.Values.pd.size | int) }}{{if .}},{{end}}pd{{.}}=http://pd{{.}}:2380{{- end }}
+{{- range until (.Values.pd.size | int) }}{{if .}},{{end}}pd{{.}}=http://{{if eq $.Values.networkMode "host"}}127.0.0.1:{{add (add ($.Values.pd.port | int) 10000) .}}{{else}}pd{{.}}:2380{{end}}{{- end }}
```
Comment (onlymellb) [inline compose/templates/docker-compose.yml]: This code is too long and difficult to read, I think the following format is clearer. 
```
{{- define "initial_cluster" }}
  {{- range until (.Values.pd.size | int) }}
    {{- if . -}}
      ,
    {{- end -}}
      pd{{ . }}=http://
    {{- if eq $.Values.networkMode "host" -}}
      127.0.0.1:{{ add (add ($.Values.pd.port | int) 10000) . }}
    {{- else -}}
      pd{{ . }}:2380
    {{- end -}}
  {{- end }}
{{- end }}
```

File: compose/templates/docker-compose.yml [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
 {{- end }}
 {{- define "pd_list" }}
-{{- range until (.Values.pd.size | int) }}{{if .}},{{end}}pd{{.}}:2379{{- end }}
+{{- range until (.Values.pd.size | int) }}{{if .}},{{end}}{{if eq $.Values.networkMode "host"}}127.0.0.1:{{add ($.Values.pd.port | int) .}}{{else}}pd{{.}}:2379{{end}}{{- end }}
```
Comment (onlymellb) [inline compose/templates/docker-compose.yml]: ditto

File: compose/values.yaml [hunk 4]
```diff
@@ -1,25 +1,35 @@
-data_dir: ./data
+dataDir: ./data
+# supported networkMode: bridge | host
+networkMode: host
```
Comment (onlymellb) [inline compose/values.yaml]: I think this var should be set to `bridge` by default

File: compose/templates/docker-compose.yml (hunk 1 above)
Comment (tennix) [reply compose/templates/docker-compose.yml]: Some users do not need monitor, like TiSpark integration test. It's meaningless to add these default values

Comment (onlymellb) [reply compose/templates/docker-compose.yml]: Maybe adding a monitor switch to do this is more appropriate?

Comment (tennix) [reply compose/templates/docker-compose.yml]: Users can comment monitor components in values.yaml to disable monitor

Comment (onlymellb) [reply compose/templates/docker-compose.yml]: In this case, users need to comment out all the monitoring related components, which may miss some components, and doing this by adding a switch will make disable monitor more simple.

File: README.md [hunk 5]
```diff
@@ -76,6 +76,21 @@ You can build docker image yourself for development test.
...
+
+When using TiKV directly without TiDB, host network mode must be enabled. This way all services use host network without isolation. So you can access all services on the host machine.
+
+You can enable this mode by setting `networkMode: host` in compose/values.yaml and regenerate docker-compose.yml. When in this mode, prometheus address in configuration files should be changed from `prometheus:9090` to `127.0.0.1:9090`, and pushgateway address should be changed from `pushgateway:9091` to `127.0.0.1:9091`.
+
+These modification can be done by:
+```bash
+# Note: this only needed when networkMode is `host`
+sed -i 's/pushgateway:9091/127.0.0.1:9091/g' config/*
+sed -i 's/prometheus:9090/127.0.0.1:9090/g' config/*
```
Comment (onlymellb) [inline README.md]: This command is not working under macOS. You should use `sed -i="" 's/xxx/yyy/g' file` to be compatible with macOS and linux.

Comment (onlymellb) [inline README.md]: In the macOS should use `sed -i "" 's/xxx/yyy/g' files` instead.

Comment (c4pt0r) [reply README.md]: Have you guys tested this on Macos?

Comment (tennix) [reply README.md]: host network mode is useless on Mac @c4pt0r

**Pull request merged**